### PR TITLE
Issue 23: Fixes submit when for Mult. Associations

### PR DESCRIPTION
**Pull request by DiegoPino**
See #23. Also includes previous changes for #14 since they are dependant. 

Tags where not being saved, even when
correctly appended because the submit functions was iterating over all
associations for a DSID. Since the use case involved a very different
FORM, that had fields not present in all others, data was not being
passed to the XML document.
Also basic cleanup of iwi_set_form_element_byhash() was done to match
for `tag` the exact logic of `tab`and this way allow better
understanding.

Finally includes all changes to allow a exact targeting of a non numeric index special element, like tab or tag, even nested.

### CODE REVIEW

**Comment by patdunlavey** [comment]
I've merged this locally and will test over the next day or so. Thanks so much for your efforts on this!

I believe this pull request supersedes https://github.com/commonmedia/islandora_webform/pull/18, so that one should be closed. Is that right @DiegoPino ?

**Comment by DiegoPino** [comment]
Right, closing #18 and linking to this. Thanks!

File: submodules/islandora_webform_ingest/includes/utilities.inc [hunk 1]
```diff
@@ -70,9 +70,9 @@ function iwi_ingestible_datastreams(FedoraObject $object, $mimetypes = array('te
                 'mime' => $dsmime,
               );
               $ingestible_dsids[$ds] = $ds;
+              // Only map the datastream once!
```
Comment (patdunlavey) [inline submodules/islandora_webform_ingest/includes/utilities.inc]: This change fixed https://github.com/commonmedia/islandora_webform/issues/25

**Comment by patdunlavey** [comment]
Accepting this pull request will resolve:
https://github.com/commonmedia/islandora_webform/issues/14
https://github.com/commonmedia/islandora_webform/issues/27

**Pull request merged**